R/ml_feature_one_hot_encoder.R

```diff
@@ -20,60 +20,119 @@ ft_one_hot_encoder <- function(x, input_col = NULL, output_col = NULL,
+      input_col = .args[["input_cols"]], output_col = .args[["output_cols"]], uid = .args[["uid"]]
+    ) %>%
+      invoke("setDropLast", .args[["drop_last"]]) %>%
+      new_ml_one_hot_encoder()
+  }
 
-  jobj <- spark_pipeline_stage(
-    x, "org.apache.spark.ml.feature.OneHotEncoder",
-    input_col = .args[["input_col"]], output_col = .args[["output_col"]], uid = .args[["uid"]]
-  ) %>%
-    invoke("setDropLast", .args[["drop_last"]])
-
-  new_ml_one_hot_encoder(jobj)
+  estimator
 }
 
 #' @export
-ft_one_hot_encoder.ml_pipeline <- function(x, input_col = NULL, output_col = NULL,
+ft_one_hot_encoder.ml_pipeline <- function(x, input_cols = NULL, output_cols = NULL, handle_invalid = "error",
                                            drop_last = TRUE, uid = random_string("one_hot_encoder_"), ...) {
-  stage <- ft_one_hot_encoder.spark_connection(
-    x = spark_connection(x),
-    input_col = input_col,
-    output_col = output_col,
-    drop_last = drop_last,
-    uid = uid,
-    ...
-  )
+  if(is_spark_3(spark_connection(x))) {
+    stage <- ft_one_hot_encoder.spark_connection(
+      x = spark_connection(x),
+      input_cols = input_cols,
+      output_cols = output_cols,
+      handle_invalid = handle_invalid,
+      drop_last = drop_last,
+      uid = uid,
+      ...
+    )
+  } else {
+    stage <- ft_one_hot_encoder.spark_connection(
+      x = spark_connection(x),
+      input_cols = input_cols,
+      output_cols = output_cols,
+      drop_last = drop_last,
+      uid = uid,
+      ...
+    )
+  }
+
   ml_add_stage(x, stage)
 }
 
 #' @export
-ft_one_hot_encoder.tbl_spark <- function(x, input_col = NULL, output_col = NULL,
+ft_one_hot_encoder.tbl_spark <- function(x, input_cols = NULL, output_cols = NULL, handle_invalid = "error",
                                          drop_last = TRUE, uid = random_string("one_hot_encoder_"), ...) {
-  stage <- ft_one_hot_encoder.spark_connection(
-    x = spark_connection(x),
-    input_col = input_col,
-    output_col = output_col,
-    drop_last = drop_last,
-    uid = uid,
-    ...
-  )
-  ml_transform(stage, x)
+  if(is_spark_3(spark_connection(x))) {
+    stage <- ft_one_hot_encoder.spark_connection(
+      x = spark_connection(x),
+      input_cols = input_cols,
+      output_cols = output_cols,
+      handle_invalid = handle_invalid,
+      drop_last = drop_last,
+      uid = uid,
+      ...
+    )
+  } else {
+    stage <- ft_one_hot_encoder.spark_connection(
+      x = spark_connection(x),
+      input_cols = input_cols,
+      output_cols = output_cols,
+      drop_last = drop_last,
+      uid = uid,
+      ...
+    )
+  }
+
+  if (is_ml_transformer(stage)) {
+    ml_transform(stage, x)
+  } else {
+    ml_fit_and_transform(stage, x)
+  }
 }
 
 new_ml_one_hot_encoder <- function(jobj) {
   new_ml_transformer(jobj, class = "ml_one_hot_encoder")
 }
 
+new_ml_one_hot_encoder_model <- function(jobj) {
+  spark_require_version(x, "3.0.0")
```

Comment:
Notice that `x` here is undefined, replace `x` with `spark_connection(jobj)`. Fixing this should also fix the last remaining test.